### PR TITLE
refactor(CreateCollection): to compositionApi

### PR DESCRIPTION
**Pull request by pbkompasz**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [ ] Feature
- [X] Refactoring

## Context

- [X] Related with #4750 
- [X] Related with #6392 
- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [X] My contribution builds **clean without any errors or warnings**
- [X] I've merged recent default branch -- **main** and I've no conflicts
- [X] I've tried to respect high code quality standards
- [X] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Did your issue had any of the "$" label on it?

- [X] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=16hqk8Tjugo9AwSzKkxXc1dsRK7iJetpx1YBaoXYwdHzuNAt&usdamount=50&donation=true)

#### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### <samp>🤖 Generated by Copilot at 773c66d</samp>

Refactored `CreateCollection.vue` component to use Vue 3 and Composition API, and removed unused and outdated code.

### <samp>🤖 Generated by Copilot at 773c66d</samp>

> _A component was due for a fix_
> _So it used Vue 3 and its tricks_
> _With the Composition API_
> _And no `v-bind.sync`, oh my_
> _The code is now cleaner and slicks_

### CODE REVIEW

**Comment by pbkompasz** [comment]
There were unused methods in the previous version. Should I remove those?

File: components/bsx/Create/CreateCollection.vue [hunk 1]
```diff
@@ -37,23 +37,18 @@
   </div>
 </template>
 
-<script lang="ts">
+<script setup lang="ts">
+import { ref, watch } from 'vue'
```
Comment (stephenjason89) [inline components/bsx/Create/CreateCollection.vue:41]:
```suggestion
```

you can remove this, it is auto imported

File: components/bsx/Create/CreateCollection.vue [hunk 2]
```diff
@@ -62,143 +57,153 @@ import shouldUpdate from '@/utils/shouldUpdate'
...
+const { t } = $i18n
+
+const emit = defineEmits(['created'])
+
+const base: BaseCollectionType = {
+  name: '',
+  file: null,
+  description: '',
 }
+const id = '0'
```
Comment (stephenjason89) [inline components/bsx/Create/CreateCollection.vue:82]: this is not being used

File: components/bsx/Create/CreateCollection.vue [hunk 3]
```diff
@@ -62,143 +57,153 @@ import shouldUpdate from '@/utils/shouldUpdate'
...
+const attributes: Attribute[] = []
```
Comment (stephenjason89) [inline components/bsx/Create/CreateCollection.vue:83]: this is always an empty array, do we need this?

File: components/bsx/Create/CreateCollection.vue [hunk 4]
```diff
@@ -62,143 +57,153 @@ import shouldUpdate from '@/utils/shouldUpdate'
...
+const checkValidity = () => {
+  return collectionForm.value?.checkValidity()
+}
 
-  @Watch('accountId', { immediate: true })
-  async onAccountIdChange(val: string, oldVal: string) {
-    if (shouldUpdate(val, oldVal)) {
-      this.feesToken = await getFeesToken()
-    }
+const balanceNotEnoughMessage = computed(() => {
```
Comment (stephenjason89) [inline components/bsx/Create/CreateCollection.vue:97]: const balanceNotEnoughMessage = computed(() =>
  balanceNotEnough.value ? t('tooltip.notEnoughBalance') : ''
)

File: components/bsx/Create/CreateCollection.vue [hunk 5]
```diff
@@ -62,143 +57,153 @@ import shouldUpdate from '@/utils/shouldUpdate'
...
-  public async created() {
-    onApiConnect(this.apiUrl, (api) => {
-      const classDeposit = getclassDeposit(api)
-      const metadataDeposit = getMetadataDeposit(api)
-      this.collectionDeposit = (classDeposit + metadataDeposit).toString()
-    })
-  }
-  get balanceOfToken() {
-    return getBalance(this.feesToken)
+watch(accountId, async (newVal, oldVal) => {
```
Comment (stephenjason89) [inline components/bsx/Create/CreateCollection.vue:104]: can you check if this is being executed?
I don't see anywhere in the code that accountId is being changed.

File: components/bsx/Create/CreateCollection.vue [hunk 6]
```diff
@@ -62,143 +57,153 @@ import shouldUpdate from '@/utils/shouldUpdate'
...
+const tryToEstimateTx = async () => {
+  const api = await ApiFactory.useApiInstance(apiUrl.value)
+  const cb = api.tx.utility.batchAll
+  const metadata = dummyIpfsCid()
+  const randomId = 0
+  const args = [createArgs(randomId, metadata)]
+  return estimate(accountId.value, cb, args)
+}
+
+const checkBalanceBeforeTx = async () => {
```
Comment (stephenjason89) [inline components/bsx/Create/CreateCollection.vue:135]: This is not being used either, do we still need this? Please confirm @roiLeo

File: components/bsx/Create/CreateCollection.vue [hunk 7]
```diff
@@ -62,143 +57,153 @@ import shouldUpdate from '@/utils/shouldUpdate'
...
-    if (!hasTokens) {
-      throw new Error(
-        `Not enough tokens: Currently have ${formatBalance(
-          this.balance,
-          this.decimals,
-          this.unit
-        )} tokens`
-      )
-    }
+const submit = async () => {
```
Comment (stephenjason89) [inline components/bsx/Create/CreateCollection.vue:151]: I believe this should be improved, watchers shouldn't be placed inside this function. This registers the watchers every time submit is clicked.

**Comment by stephenjason89** [comment]
> There were unused methods in the previous version. Should I remove those?

I believe so unless it is used by other components via templateRefs then you need to use `defineExpose`

File: components/bsx/Create/CreateCollection.vue (hunk 6 above)
Comment (roiLeo) [reply components/bsx/Create/CreateCollection.vue:135]: it's commented line 189, we might need this to check balance

File: components/bsx/Create/CreateCollection.vue (hunk 7 above)
Comment (pbkompasz) [reply components/bsx/Create/CreateCollection.vue:151]: Yeah, I left everything in before getting feedback. I will remove the unused methods and improve the `submit` method.

File: components/bsx/Create/CreateCollection.vue (hunk 4 above)
Comment (pbkompasz) [reply components/bsx/Create/CreateCollection.vue:97]: This will get replaced in #6738.

**Comment by stephenjason89** [comment]
This is a duplicate of #6924

**Pull request closed without merging**